crates/router/src/scheduler/consumer.rs

```diff
@@ -62,6 +77,13 @@ pub async fn start_consumer(
                 },
                 sync::Arc::clone(&consumer_operation_counter),
             ));
+
+            match rx.try_recv() {
```

Comment:
We could use this as the root match instead of if is_ready

Also add separate loggers for Ok() and receiver closed error